File: Top 100 Codes/03-Codes for Recursion/07-LCM of two numbers/LCMoftwonumbers.cpp
#include <iostream>
using namespace std;

// Function to calculate GCD using recursion
int gcd(int a, int b) {
    if (b == 0)
        return a;
    return gcd(b, a % b);
}

// Function to calculate LCM using the formula: LCM(a, b) = (a * b) / GCD(a, b)
int lcm(int a, int b) {
    return (a * b) / gcd(a, b);
}

int main() {
    int num1, num2;
    cout << "Enter two numbers: ";
    cin >> num1 >> num2;

    cout << "LCM of " << num1 << " and " << num2 << " is: " << lcm(num1, num2) << endl;

    return 0;
}

File: Top 100 Codes/03-Codes for Recursion/07-LCM of two numbers/LCMoftwonumbers.py
def gcd(a, b):
    if b == 0:
        return a
    return gcd(b, a % b)

def lcm(a, b):
    return (a * b) // gcd(a, b)

num1 = int(input("Enter first number: "))
num2 = int(input("Enter second number: "))

print(f"LCM of {num1} and {num2} is: {lcm(num1, num2)}")
